### ISSUE TITLE
[Bug] Syntax errors with workflow conditional expressions

### ISSUE THREAD

**Issue by andyfeller** [skills/connect-the-dots#14]
**Describe the bug**

1. **Invalid YAML for multi-line conditional**

    The conditional expressions used to selectively trigger workflows in this skills course appear to have syntax errors.  The following list of workflows use multi-line conditionals that are terminated improperly:
    
    - connect-the-dots/.github/workflows/1-resolve-duplicate-issues.yml
      
      ```yaml
      -49-    if: ${{ github.repository_owner != 'TBD-organization' }} 
      :50:      && needs.get_current_step.outputs.current_step == 1 }}
      ```
      
    - connect-the-dots/.github/workflows/2-find-commit-in-history.yml
      
      ```yaml
      -47-    if: ${{ github.repository_owner != 'TBD-organization' }} 
      :48:      && needs.get_current_step.outputs.current_step == 2 }}
      ```
      
    - connect-the-dots/.github/workflows/3-fix-broken-sidebar.yml
      
      ```yaml
      -44-    if: ${{ github.repository_owner != 'TBD-organization' }} 
      :45:      && needs.get_current_step.outputs.current_step == 3 }}
      ```

1. **Invalid YAML for multi-line conditional**

   Using all of the examples above, this is invalid syntax for [block YAML use cases](https://yaml.org/spec/1.2.2/#chapter-8-block-style-productions).

   Instead of:

   ```yaml
         if: ${{ github.repository_owner != 'TBD-organization'
           && needs.get_current_step.outputs.current_step == 6 }}
   ```

   One of the following forms must be used:

   1. Single line

      ```yaml
            if: ${{ github.repository_owner != 'TBD-organization' && needs.get_current_step.outputs.current_step == 6 }}
      ```

   1. Multi line block

      ```yaml
            if: |
              ${{ github.repository_owner != 'TBD-organization'
                  && needs.get_current_step.outputs.current_step == 6 }}
      ```

**To Reproduce**

**Expected behavior**
Use of this skills course respected the state of the workflow step under execution without errors.

**Screenshots**

**Device information**
 - Type: [e.g. iPhone6]

**Additional context**

**Comment by andyfeller** [comment]
cc: @emilyistoofunky

**Comment by sinsukehlab** [comment]
I think my PR (#10) will solve this issue.

**Comment by sinsukehlab** [comment]
I tried multi-line conditional expressions you suggested, but they did not work well.
single-line conditional expressions will work.

**Comment by andyfeller** [comment]
> I tried multi-line conditional expressions you suggested, but they did not work well. single-line conditional expressions will work.

@sinsukehlab : could you link to the specific action workflow runs where the multi line conditional was being used and the error, please?

I would like to follow up internally to understand what about the suggestion was off to avoid future wasted effort! 🙇

**Comment by sinsukehlab** [comment]
I really appreciate your kind reply. :smile:
To my regret, I have deleted the repository that contains the multi-line conditionals. :bow:
I describe what happened instead.

**To Reproduce**
A YAML multi-line conditional sample to reproduce the behavior:
```yaml
    if: |
      ${{ !github.event.repository.is_template
          && needs.get_current_step.outputs.current_step == 0 }}
```

**Expected behavior**
The job should run only if `!github.event.repository.is_template && needs.get_current_step.outputs.current_step == 0`.

**Actual behavior**
The job runs even if `needs.get_curent_step.outputs.current_step != 0`.

**Comment by sinsukehlab** [comment]
I think you should use the folded style (`>-`) instead of the literal style (`|`).
Cf. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#status

Correct multi-line conditional sample would go:
```yaml
    if: >-
      ${{ !github.event.repository.is_template
          && needs.get_current_step.outputs.current_step == 0 }}
```

I think this is because `|` inserts a LF (`\n`) while `>-` inserts a space.

**Comment by andyfeller** [comment]
This appears as if it was fixed in #10 , so closing it for now
